### PR TITLE
feat: add html link component

### PR DESCRIPTION
**Pull request by hamza14khan**
- Addition of html link component and macro.
- Customisation options with `external`, `noUnderline`, `dark` and `noVisited` states.
- Updated wagtail example with link usage.

TODOs: 
- Add custom class for `'text-decoration-skip-ink': 'none'`.
- Font smoothing adjustments on create theme `font-smoothing: antialiased;` 

**Storybook Testing**
Screenshot 2024-08-30 at 16 33 51

**Wagtail Example**
Screenshot 2024-08-29 at 10 36 48

### CODE REVIEW

**Review by pregno — COMMENTED** [review]
Could you add a simple usage example in `example/wagtail` and `example/nextjs`?

**Review by BogdanDanilescu — APPROVED** [review]

File: packages/html/ds/src/link/link.html [hunk 1]
```diff
@@ -0,0 +1,52 @@
+{% macro govieLink(props) %}
+  {% macro getDarlClass(dark) %}
```
Comment (pregno) [inline packages/html/ds/src/link/link.html:2]:
```suggestion
  {% macro getDarkClass(dark) %}
```

File: packages/html/ds/src/link/link.html [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      rel="noreferrer noopener" target="_blank"
+    {% endif %}
+  {% endmacro %}
+  <a
+    href="{{ props.href }}"
+    {{ getExternalAttribute(props.external) | trim }}
+    class="
+      {{ getUnderlineClass(props.noUnderline) | trim }}
+      {{ getVisitedClass(props.noVisited, props.dark) | trim }}
+      {{ getDarlClass(props.dark) | trim }}
```
Comment (pregno) [inline packages/html/ds/src/link/link.html:40]:
```suggestion
      {{ getDarkClass(props.dark) | trim }}
```

**Review by pregno — CHANGES_REQUESTED** [review]

File: packages/html/ds/src/link/link.html (hunk 1 above)
Comment (hamza14khan) [reply packages/html/ds/src/link/link.html:2]: Good catch!

File: packages/html/ds/src/link/link.html [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      hover:gi-text-blue-800
```
[the remark this answers is not in the document]
Comment (hamza14khan) [reply packages/html/ds/src/link/link.html:41]: Yes, spot on! Its duplicate. Updated ✅

**Review by pregno — APPROVED** [review]

**Pull request merged**